src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx

```diff
@@ -106,6 +114,23 @@ export class DataSourceSelectable extends React.Component<
             ...this.state,
             dataSourceOptions,
           });
+
+          const selectedDataSource = getDefaultDataSource(
+            filteredDataSources,
+            LocalCluster,
+            this.props.uiSettings,
+            this.props.hideLocalCluster,
+            this.props.selectedOption
+          );
+          if (selectedDataSource.length === 0) {
+            this.props.notifications.addWarning('No connected data source available.');
+          } else {
+            this.setState({
+              ...this.state,
+              selectedOption: selectedDataSource,
+            });
+            this.props.onSelectedDataSource(selectedDataSource);
```

Comment:
Should it be `onSelectedDataSources`?
```suggestion
            this.props.onSelectedDataSources(selectedDataSource);
```